ratpack-core/src/main/java/ratpack/server/internal/NettyHandlerAdapter.java

```diff
@@ -114,6 +114,7 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
 
   private void newRequest(ChannelHandlerContext ctx, HttpRequest nettyRequest) throws Exception {
     if (!nettyRequest.decoderResult().isSuccess()) {
+      LOGGER.warn("Failed to decode HTTP request.", nettyRequest.decoderResult().cause());
```

Comment:
Many times a bad request isn't something you want to warn on can we keep this at debug, users will be able to opt in if they want still.

Reply:
Agree, but I would be fine with warn without stacktrace.